lib/ruby_lsp/requests/support/sorbet.rb

```diff
@@ -0,0 +1,126 @@
+              "noreturn" => Annotation.new(arity: 0, receiver: true),
+              "requires_ancestor" => Annotation.new(arity: 0),
+              "reveal_type" => Annotation.new(arity: 1, receiver: true),
+              "sealed!" => Annotation.new(arity: 0),
+              "self_type" => Annotation.new(arity: 0, receiver: true),
+              "sig" => Annotation.new(arity: 0),
+              "type_member" => Annotation.new(arity: (0..1)),
+              "type_template" => Annotation.new(arity: 0),
+              "unsafe" => Annotation.new(arity: 1),
+              "untyped" => Annotation.new(arity: 0, receiver: true),
+            },
+            T::Hash[String, Annotation],
+          )
+
+          sig do
+            params(
+              node: T.any(SyntaxTree::CallNode, SyntaxTree::VCall),
+            ).returns(T::Boolean)
+          end
+          def annotation?(node)
+            annotation = annotation(node)
+
+            return false if annotation.nil?
+
+            return false unless annotation.supports_receiver?(receiver_name(node))
+
+            annotation.supports_arity?(node.arity)
+          end
+
+          private
+
+          sig { params(node: T.any(SyntaxTree::CallNode, SyntaxTree::VCall)).returns(T.nilable(Annotation)) }
+          def annotation(node)
+            case node
+            when SyntaxTree::VCall
+              ANNOTATIONS[node.value.value]
+            when SyntaxTree::CallNode
+              ANNOTATIONS[node.message.value]
+            else
+              T.absurd(node)
+            end
+          end
+
+          sig do
+            params(receiver: T.any(SyntaxTree::CallNode, SyntaxTree::VCall)).returns(T.nilable(String))
+          end
+          def receiver_name(receiver)
+            case receiver
+            when SyntaxTree::CallNode
+              node_name(receiver.receiver)
+            when SyntaxTree::VCall
+              nil
+            else
+              T.absurd(receiver)
+            end
+          end
+
+          sig do
+            params(node: T.nilable(T.any(
+              SyntaxTree::VarRef,
+              SyntaxTree::CallNode,
+              SyntaxTree::VCall,
+              SyntaxTree::Ident,
+              SyntaxTree::Backtick,
+              SyntaxTree::Const,
+              SyntaxTree::Op,
+              Symbol,
+            ))).returns(T.nilable(String))
+          end
+          def node_name(node)
+            case node
+            when NilClass
+              nil
+            when SyntaxTree::VarRef
+              node.value.value
+            when SyntaxTree::CallNode
+              node_name(node.receiver)
+            when SyntaxTree::VCall
+              node_name(node.value)
+            when SyntaxTree::Ident
+              node.value
+            when SyntaxTree::Backtick
+              node.value
+            when SyntaxTree::Const
+              node.value
+            when SyntaxTree::Op
+              node.value
```

Comment:
NIT: can we merge these into the same `when` branch since they use the same method?